src/utils/spawning.rs

```diff
@@ -141,6 +141,24 @@ fn spawn_sync(
         process.env("DESKTOP_STARTUP_ID", token.as_str());
     }
 
+    // In `main.rs`, we set a signal handler with `calloop::signals::Signals::new`.
+    // This does two things:
+    // 1. It blocks the thread from receiving these signals normally (pthread_sigmask)
+    // 2. It creates a signalfd to read them in the event loop.
+    //
+    // When spawning children, the signalfd is trivial: `Signals::new` creates it with CLOEXEC.
+    //
+    // But, the sigmask is always inherited, so we want to clear it before spawning children.
+    // That way, we don't affect their normal signal handling. In particular, if a child doesn't
+    // care about signals, we must not block it from receiving them.
+    unsafe {
+        process.pre_exec(|| {
+            nix::sys::signalfd::SigSet::empty()
+                .thread_set_mask()
+                .map_err(Into::into)
```

Comment:
I'm handling the error here instead of discarding it, because as far as i can tell, the only errors it can ever return is from incorrect usage. So, there is no error case that *can* occur here which we would even want to ignore.